src/ai_sys.rs

```diff
@@ -0,0 +1,596 @@
+            let x = player_transform.translation.x - enemy_transform.translation.x;
+            let target_angle = atan2f(y, x);
+
+            let angle_between = enemy_transform
+                .rotation
+                .angle_between(Quat::from_rotation_z(target_angle))
+                - PI / 2.0;
+
+            turn_toward(&mut enemy_transform, enemy_ship.turn_speed, angle_between);
+
+            if (-0.15 < angle_between)
+                && (angle_between < 0.15)
+                && enemy_ship.primary_weapon.cd_timer.finished()
+            {
+                // ######### FIRE! ##########
+                // The projectile's transform should originate from the firing ship.
+                let mut projectile_transform = Transform::from_xyz(
+                    enemy_transform.translation.x,
+                    enemy_transform.translation.y,
+                    0.0,
+                )
+                .with_scale(GLOBAL_RESCALE_V);
+                // Modify it a little so that it originates from just in front of the firing ship.
+                projectile_transform.translation += enemy_transform.up() * 75.0 * GLOBAL_RESCALE_V;
+                // Ensure that it is rotated in a way that aligns with the firing ship.
+                projectile_transform.rotation = enemy_transform.rotation.clone();
+                commands.spawn((
+                    SpriteBundle {
+                        transform: projectile_transform,
+                        texture: asset_server.load(enemy_ship.primary_weapon.sprite_path.clone()),
+                        ..default()
+                    },
+                    // The Projectile is granted value's from the ship's primary_weapon component.
+                    // This depends on the type of projectile the cannon fires.
+                    enemy_ship
+                        .primary_weapon
+                        .fire(projectile_transform.up(), vel.velocity.length()),
+                ));
+            }
+        }
+    }
+}
+
+pub fn speedy_ai_system(
+    mut commands: Commands,
+    mut q_enemy: Query<
+        (
+            &mut Ship,
+            &mut Transform,
+            &mut Velocity,
+            &mut AITimer,
+            &mut AITimer2,
+            &Mass,
+            &Thruster,
+        ),
+        (With<Enemy>, With<SpeedyAI>, Without<Player>),
+    >,
+    q_player: Query<(&Transform), (With<Player>, Without<Enemy>)>,
+    asset_server: Res<AssetServer>,
+    time: Res<Time>,
+) {
+    // The intended behavior of the "speedy" enemy is to fly into mid-range of the player.
+    // Once mid-range, alternatively fire on the player and fly away to change position.
+    // Can be viewed as a FSM.
+    // State 1: Fly toward player.
+    // State 2: Turn and Fire at Player.
+    // State 3: Turn and briefly engage thrusters to reposition.
+    // If far from player, enter state 1. Otherwiser alternate states 2 and 3 every few seconds.
```

Comment:
More complex AI like this is a great candidate for state machines. Create an enum to keep track of the current state (for example: "idle", "observing player", "flying towards player", "shooting", etc.) and then you can easily apply different logic depending on the current state, and transition between states as appropriate.